### PR TITLE
Revert "Convert pkg/api to Typescript (#853)"

### PR DESCRIPTION
**Pull request by mturley**
This reverts #853.

This change appears to have broken Cachito behavior during builds. Until we can determine a better fix, reverting this so we don't block 2.3 builds while some of us are away for year-end recharge.

@gildub after the holidays, let's revisit #853 and try to implement it again with some more scrutiny and make sure we involve Franco and make sure end-to-end builds are still working.

cc @seanforyou23 @fbladilo 

Cachito debug log apparently caused by the change:

```
[2021-12-22 17:56:35,350 cachito.workers.tasks.utils DEBUG utils.get_request_state] Getting the state of request 187413
[2021-12-22 17:56:35,460 cachito.workers.tasks.general INFO general.fetch_app_source] Fetching the source from "https://github.com/konveyor/forklift-ui" at reference "a22e5350ab8964550ce67dec26a8a8d50a244c0e"
[2021-12-22 17:56:35,460 cachito.workers.tasks.utils INFO utils.set_request_state] Setting the state of request 187413 to "in_progress" with the reason "Fetching the application source"
[2021-12-22 17:56:35,595 cachito.workers.scm DEBUG scm.repo_name] Parsed the repository name "konveyor/forklift-ui" from https://github.com/konveyor/forklift-ui
[2021-12-22 17:56:35,595 cachito.workers.paths DEBUG paths.__new__] Ensure directory /var/lib/cachito/sources/konveyor/forklift-ui exists.
[2021-12-22 17:56:35,597 cachito.workers.scm DEBUG scm.fetch_source] The archive already exists at "/var/lib/cachito/sources/konveyor/forklift-ui/a22e5350ab8964550ce67dec26a8a8d50a244c0e.tar.gz"
[2021-12-22 17:56:35,597 cachito.workers.scm DEBUG scm._verify_archive] Verifying the archive at /var/lib/cachito/sources/konveyor/forklift-ui/a22e5350ab8964550ce67dec26a8a8d50a244c0e.tar.gz
[2021-12-22 17:56:36,302 cachito.workers.paths DEBUG paths.__new__] Ensure directory /var/lib/cachito/bundles/temp/187413 exists.
[2021-12-22 17:56:36,302 cachito.workers.paths DEBUG paths.__new__] Ensure directory /var/lib/cachito/bundles/temp/187413/deps exists.
[2021-12-22 17:56:36,304 cachito.workers.tasks.general DEBUG general.fetch_app_source] Extracting /var/lib/cachito/sources/konveyor/forklift-ui/a22e5350ab8964550ce67dec26a8a8d50a244c0e.tar.gz to /var/lib/cachito/bundles/temp/187413
[2021-12-22 17:56:37,104 celery.app.trace INFO trace.info] Task cachito.workers.tasks.general.fetch_app_source[a6c05ba1-838b-4ed0-8972-cbf1d4d29868] succeeded in 1.7550894857849926s: None
[2021-12-22 17:56:37,107 cachito.workers.tasks.utils DEBUG utils.get_request_state] Getting the state of request 187413
[2021-12-22 17:56:37,752 cachito.workers.tasks.npm INFO npm.fetch_npm_source] npm version: 8.0.0
[2021-12-22 17:56:38,647 cachito.workers.nexus INFO nexus.execute_script] Executing the Nexus script js_cleanup
[2021-12-22 17:56:38,874 cachito.workers.nexus INFO nexus.execute_script] The Nexus script js_cleanup executed successfully
[2021-12-22 17:56:38,874 celery.app.trace INFO trace.info] Task cachito.workers.tasks.npm.cleanup_npm_request[0f587196-dec8-442a-a52b-910de31d09ce] succeeded in 0.2280062970239669s: None
UG npm.fetch_npm_source] Checking if the application source uses npm
[2021-12-22 17:56:37,779 cachito.workers.tasks.npm DEBUG npm._verify_npm_files] Ensuring there is no node_modules directory present
[2021-12-22 17:56:37,779 cachito.workers.tasks.npm INFO npm.fetch_npm_source] Configuring Nexus for npm for the request 187413
[2021-12-22 17:56:37,779 cachito.workers.tasks.utils INFO utils.set_request_state] Setting the state of request 187413 to "in_progress" with the reason "Configuring Nexus for npm"
[2021-12-22 17:56:37,914 cachito.workers.nexus INFO nexus.execute_script] Executing the Nexus script js_before_content_staged
[2021-12-22 17:56:38,051 cachito.workers.nexus INFO nexus.execute_script] The Nexus script js_before_content_staged executed successfully
[2021-12-22 17:56:38,051 cachito.workers.tasks.npm INFO npm.fetch_npm_source] Fetching the npm dependencies for request 187413 in subpath .
[2021-12-22 17:56:38,051 cachito.workers.tasks.utils INFO utils.set_request_state] Setting the state of request 187413 to "in_progress" with the reason "Fetching the npm dependencies at the "." directory""
[2021-12-22 17:56:38,172 cachito.workers.tasks.utils DEBUG utils.get_request] Getting request 187413
[2021-12-22 17:56:38,297 cachito.workers.pkg_managers.npm INFO npm._get_deps] The dependency {'version': 'file:pkg/api', 'requires': {'@types/ejs': '^3.0.6', '@types/express': '^4.17.12', '@types/simple-oauth2': '^4.1.0', 'ejs': '^3.1.6', 'express': '^4.17.1', 'get-ssl-certificate': '^2.3.3', 'http-proxy-middleware': '^2.0.0', 'simple-oauth2': '^4.2.0'}} is not from the npm registry
[2021-12-22 17:56:38,297 cachito.workers.tasks.npm ERROR npm.fetch_npm_source] Failed to fetch npm dependencies for request 187413
Traceback (most recent call last):
  File "/src/cachito/workers/tasks/npm.py", line 143, in fetch_npm_source
    package_and_deps_info = resolve_npm(
  File "/src/cachito/workers/pkg_managers/npm.py", line 296, in resolve_npm
    package_and_deps_info = get_package_and_deps(package_json_path, package_lock_path)
  File "/src/cachito/workers/pkg_managers/npm.py", line 216, in get_package_and_deps
    name_to_deps, top_level_replacements = _get_deps(
  File "/src/cachito/workers/pkg_managers/npm.py", line 77, in _get_deps
    nexus_replacement = convert_to_nexus_hosted(name, info)
  File "/src/cachito/workers/pkg_managers/npm.py", line 142, in convert_to_nexus_hosted
    dep_in_nexus = process_non_registry_dependency(dep)
  File "/src/cachito/workers/pkg_managers/general_js.py", line 625, in process_non_registry_dependency
    raise CachitoError(
cachito.errors.CachitoError: The dependency forklift-ui-api@file:pkg/api is hosted in an unsupported location
[2021-12-22 17:56:38,298 cachito.workers.tasks.utils INFO utils.set_request_state] Setting the state of request 187413 to "failed" with the reason "The dependency forklift-ui-api@file:pkg/api is hosted in an unsupported location"
[2021-12-22 17:56:38,666 celery.app.trace ERROR trace._log_error] Task cachito.workers.tasks.npm.fetch_npm_source[8259b05a-abb3-48d4-b425-c5794db985fa] raised unexpected: CachitoError('The dependency forklift-ui-api@file:pkg/api is hosted in an unsupported location')
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 405, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 697, in __protected_call__
    return self.run(*args, **kwargs)
  File "/src/cachito/workers/tasks/utils.py", line 145, in task_with_state_check
    return task_fn(*args, **kwargs)
  File "/src/cachito/workers/tasks/npm.py", line 143, in fetch_npm_source
    package_and_deps_info = resolve_npm(
  File "/src/cachito/workers/pkg_managers/npm.py", line 296, in resolve_npm
    package_and_deps_info = get_package_and_deps(package_json_path, package_lock_path)
  File "/src/cachito/workers/pkg_managers/npm.py", line 216, in get_package_and_deps
    name_to_deps, top_level_replacements = _get_deps(
  File "/src/cachito/workers/pkg_managers/npm.py", line 77, in _get_deps
    nexus_replacement = convert_to_nexus_hosted(name, info)
  File "/src/cachito/workers/pkg_managers/npm.py", line 142, in convert_to_nexus_hosted
    dep_in_nexus = process_non_registry_dependency(dep)
  File "/src/cachito/workers/pkg_managers/general_js.py", line 625, in process_non_registry_dependency
    raise CachitoError(
cachito.errors.CachitoError: The dependency forklift-ui-api@file:pkg/api is hosted in an unsupported location
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 FROM registry.access.redhat.com/ubi8/nodejs-16 as builder
 COPY . .
 USER root
-RUN npm config set unsafe-perm true && npm run install:ws && npm run build
```
Comment (seanforyou23) [inline Dockerfile:5]: Maybe just verify with @gildub if we should re-introduce this change after this revert goes through?

**Review by seanforyou23 — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (mturley) [reply Dockerfile:5]: hmm, good call. I'll open an issue

Comment (mturley) [reply Dockerfile:5]: https://github.com/konveyor/forklift-ui/issues/872

**Comment by mturley** [comment]
Reopened https://github.com/konveyor/forklift-ui/issues/760 so we don't forget to revisit this after the revert.

**Pull request merged**